src/eventListeners/index.ts

```diff
@@ -0,0 +1,64 @@
+import { verbose } from '~utils';
+
+import { EventListener, EventListenerType } from './types';
+
+export * from './types';
+export * from './colony';
+export * from './network';
+export * from './votingReputation';
+export * from './extensions';
+export * from './token';
+
+let listeners: EventListener[] = [];
+
+export const getEventListeners = (): EventListener[] => listeners;
+
+export const setEventListeners = (newListeners: EventListener[]): void => {
+  listeners = newListeners;
+};
+
+export const addEventListener = (listener: EventListener): void => {
+  verbose(
+    `Added listener for event ${listener.eventSignature}`,
+    listener.address ? `filtering address ${listener.address}` : '',
+  );
+  listeners.push(listener);
+};
+
+export const getMatchingListener = (
+  logTopics: string[],
+  logAddress: string,
+): EventListener | null => {
+  return (
+    listeners.find((listener) => {
+      if (listener.address && logAddress !== listener.address) {
+        return false;
+      }
+
+      return listener.topics.every((topic, index) => {
+        if (topic === null) {
+          // if listener topic is null, skip the check
+          return true;
+        }
+
+        return topic === logTopics[index];
+      });
+    }) ?? null
+  );
+};
+
+export const getListenersStats = (): string => JSON.stringify(listeners);
+
+export const getListenerContractEventProperties = (
```

Comment:
Nice solution to the case of needing the colony address inside VotingRep specific handlers